### PR TITLE
feat: LambdaSstoreTransaction

### PR DESCRIPTION
**Pull request by ivaylonikolov7**
**Description**:
implements LambdaSstoreTransaction related to Hiero Hooks

**Related issue(s)**:
#3453 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/hooks/LambdaSStoreTransaction.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    /**
+     *
+     * @param {object} props
+     * @param {import("../hooks/HookId.js").default} [props.hookId]
+     * @param {import("../hooks/LambdaStorageUpdate.js").LambdaStorageUpdate[]} [props.storageUpdates]
+     */
+    constructor(props = {}) {
+        super();
+
+        this.hookId = null;
```
Comment (venilinvasilev) [inline src/hooks/LambdaSStoreTransaction.js]: nit: Don't we use getters and mark the properties as internal e.g. this._hookId. This applies for elsewhere occurences

File: src/hooks/LambdaSStoreTransaction.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        this.hookId = hookId;
+        return this;
+    }
+
+    /**
+     *
+     * @param {LambdaStorageUpdate[]} storageUpdates
+     * @returns {this}
+     */
+    setStorageUpdates(storageUpdates) {
```
Comment (0xivanov) [inline src/hooks/LambdaSStoreTransaction.js:75]: addStorageUpdate is missing

File: test/integration/LambdaSStoreIntegrationTest.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                .sign(accountKey)
+        ).execute(env.client);
+
+        const receipt = await response.getReceipt(env.client);
+
+        // Then: the storage is updated successfully
+        expect(receipt.status).toEqual(Status.Success);
+    });
+
+    // TODO: this shouldnt fail, only when updating according to protobufs
```
Comment (0xivanov) [inline test/integration/LambdaSStoreIntegrationTest.js]: I guess it should fail with invalid signature

File: test/integration/LambdaSStoreIntegrationTest.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+            })
+                .freezeWith(env.client)
+                .sign(invalidKey)
+                .execute(env.client);
+            await response.getReceipt(env.client);
+        } catch (error) {
+            status = error.status;
+        }
+
+        console.log(status);
```
Comment (0xivanov) [inline test/integration/LambdaSStoreIntegrationTest.js]: log

File: test/integration/LambdaSStoreIntegrationTest.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                }).execute(env.client)
+            ).getReceipt(env.client);
+        } catch (error) {
+            console.log(error);
+            status = error.status;
+        }
+
+        expect(status).toEqual(Status.HookNotFound);
+    });
+
```
Comment (0xivanov) [inline test/integration/LambdaSStoreIntegrationTest.js:208]: can add test for `TOO_MANY_LAMBDA_STORAGE_UPDATES` - the max is 10, so we'd need 11 updates

File: test/unit/LambdaSStoreTransaction.js [hunk 6]
```diff
@@ -0,0 +1,647 @@
...
+                    TransactionId.withValidStart(accountId, timestamp),
+                )
+                .freeze();
+
+            const bytes = transaction.toBytes();
+            const deserializedTx = Transaction.fromBytes(bytes);
+
+            expect(deserializedTx).to.be.an.instanceof(LambdaSStoreTransaction);
+            expect(deserializedTx.storageUpdates).to.have.lengthOf(1);
+            expect(deserializedTx.storageUpdates[0]).to.be.an.instanceof(
```
Comment (0xivanov) [inline test/unit/LambdaSStoreTransaction.js:479]: can compare what's inside of the storage update after deserialization

File: test/unit/LambdaSStoreTransaction.js [hunk 7]
```diff
@@ -0,0 +1,647 @@
...
+                    TransactionId.withValidStart(accountId, timestamp),
+                );
+
+            expect(transaction).to.be.an.instanceof(LambdaSStoreTransaction);
+            expect(transaction.hookId).to.equal(hookId);
+            expect(transaction.storageUpdates).to.have.lengthOf(1);
+        });
+    });
+
+    describe("edge cases", function () {
```
Comment (0xivanov) [inline test/unit/LambdaSStoreTransaction.js:560]: add test for from/to bytes with empty transaction - nothing set

File: src/hooks/LambdaSStoreTransaction.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+     * @returns {HookId | null}
+     */
+    get hookId() {
+        return this._hookId;
+    }
+
+    /**
+     * @returns {LambdaStorageUpdate[] | []}
+     */
+    get storageUpdates() {
```
Comment (0xivanov) [inline src/hooks/LambdaSStoreTransaction.js]: nit: move this above `setStorageUpdates`
